### PR TITLE
Agent card no longer uses meaningless reset feature to forge

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
* Agent card no longer uses meaningless reset feature to forge
* globalise the hop code proc

## Why It's Good For The Game
resetting an agent card resets everything to `initial(vars)`, but those variables do not have initial values, making them null values. This is a meaningless feature.

also, improves the forging feature more user-friendly, so you'll forge a card flawlessly.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image
image
image
image

gets Bartender automatically from selected card style

-----------

image
image
image

sets a custom title HoP here
(also, not hop, becomes Hop due to caplisation proc)

---------------

image
image
image

gets a custom title HoP from previous move instead of getting Mime title

I can check that I can flawlessly forge.

</details>

## Changelog
:cl:
code: moved some HoP card painter feature to global proc
del: Reset feature in agent cards has been removed - it does actually nothing.
tweak: Agent card forging in the user experience aspect is now greatly improved - You'll forge a card flawlessly.
/:cl:

### CODE REVIEW

File: code/__HELPERS/grammar.dm [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// Returns a list in plain english as a string
+/proc/english_list(list/input, nothing_text = "nothing", and_text = " and ", comma_text = ", ", final_comma_text = "" )
```
Comment (Archanial) [inline code/__HELPERS/grammar.dm]: moving it is pointless, it fits helpers for lists

File: code/__HELPERS/grammar.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+				if (index == total - 1)
+					comma_text = final_comma_text
+
+				output += "[input[index]][comma_text]"
+				index++
+
+			return "[output][and_text][input[index]]"
+
+/// Makes the first letter of a string to capital
+/proc/replace_first_letter_to_capital(string)
```
Comment (Archanial) [inline code/__HELPERS/grammar.dm]: do we not have some strings.dm helpers?

Comment (EvilDragonfiend) [reply code/__HELPERS/grammar.dm]: we don't have such dm file. I think you're talking about `pronouns.dm` and that file only has verb transformation.

File: code/__HELPERS/grammar.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/__HELPERS/grammar.dm]: I believe this makes sense to be here, and I am going to add more grammar correction procs there. It's a proc using lists indeed, but its usage is closer to the grammar purpose than the list purpose.
For example, if you need to use `english_list()` as a first-contributer, you'd expect the proc is related to grammar stuff, not in the list helper which is mainly used for sorting or controlling its values.
I mean, you wouldn't expect `english_list()` to be in the list helper.

File: code/game/objects/items/cards_ids.dm [hunk 3]
```diff
@@ -485,27 +487,26 @@ update_label("John Doe", "Clowny")
 				else
 					input_name = "[pick(GLOB.first_names)] [pick(GLOB.last_names)]"
 
-			var/target_occupation = stripped_input(user, "What occupation would you like to put on this card?\nNote: This will not grant any access levels other than Maintenance.", "Agent card job assignment", assignment ? assignment : JOB_NAME_ASSISTANT, MAX_MESSAGE_LEN)
-			if(!target_occupation)
-				return
-
 			// copy paste from HoP card painting machine
```
Comment (itsmeow) [inline code/game/objects/items/cards_ids.dm]: while you're here, might as well make this into a global proc

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
changed as itsmeow reviewed.

**Comment by EvilDragonfiend** [comment]
I took a look into helpers again, and I see there's one in `text.dm` I will replace it soon.

File: code/__HELPERS/grammar.dm (hunk 2 above)
Comment (EvilDragonfiend) [reply code/__HELPERS/grammar.dm]: found out it was in text.dm

File: code/__HELPERS/grammar.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/__HELPERS/grammar.dm]: now moved to `text.dm`

Comment (EvilDragonfiend) [reply code/__HELPERS/grammar.dm]: maybe out of the scope for the main topic of this PR. I will revert the relocation.

**Review by Archanial — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]
